src/main/java/org/carlmontrobotics/subsystems/Limelight.java

```diff
@@ -0,0 +1,90 @@
+package org.carlmontrobotics.subsystems;
+
+import static org.carlmontrobotics.Constants.Limelightc.*;
+import static org.carlmontrobotics.Constants.Limelightc.Apriltag.*;
+
+
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.estimator.SwerveDrivePoseEstimator;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Pose3d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.networktables.NetworkTable;
+import edu.wpi.first.networktables.NetworkTableEntry;
+import edu.wpi.first.networktables.NetworkTableInstance;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+public class Limelight extends SubsystemBase {
+  private final Drivetrain drivetrain;
+  private final SwerveDrivePoseEstimator poseEstimator;
+  private double tv, tx;
+  //private double[] rawBotPose = null;
+  private double[] targetPose = null;
+  private Pose3d botPose;
+
+  //private double distOffset, horizOffset;
+  //private double horizHeadingError, horizAdjust;
+
+  public Limelight(Drivetrain drivetrain) {
+    this.drivetrain = drivetrain;
+    poseEstimator = new SwerveDrivePoseEstimator(
+      drivetrain.getKinematics(), 
+      Rotation2d.fromDegrees(drivetrain.getHeading()), 
+      drivetrain.getModulePositions(), 
+      new Pose2d());
+    
+    //rawBotPose = NetworkTableInstance.getDefault().getTable(SHOOTER_LL_NAME).getEntry("botpose").getDoubleArray(new double[7]);
+    //targetPose = table.getEntry("targetpose_fieldspace").getDoubleArray(new double[7]);
+  }
+
+  @Override
+  public void periodic() {
+    poseEstimator.update(Rotation2d.fromDegrees(drivetrain.getHeading()), drivetrain.getModulePositions());
+    updateBotPose3d();
+    getDistanceToTargetSpeaker();
+    getCurrentPose();
+  }
+
+  public void updateBotPose3d(){
+    botPose = LimelightHelpers.getBotPose3d(SHOOTER_LL_NAME);
+  }
+
+
+  public Pose2d getCurrentPose(){
+    Pose2d estimatedPos = poseEstimator.getEstimatedPosition();
+    System.out.println(estimatedPos.getX());
+    System.out.println(estimatedPos.getY());
+    System.out.println(estimatedPos.getRotation().getDegrees() + "degrees");
+    return poseEstimator.getEstimatedPosition();
```

Comment:
Probably won't make a significant difference but return the estimated position you printed out to console, not a new estimated position.